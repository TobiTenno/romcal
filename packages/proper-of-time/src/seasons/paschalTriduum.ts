import {
  Color,
  LiturgicalDayDef,
  Period,
  Precedence,
  PROPER_OF_TIME_ID,
  Season,
} from '@romcal/shared';

/**
 * Compute the Proper of the Paschal Triduum
 * @param yearOffset A Liturgical year offset
 */
export const paschalTriduum = (yearOffset = 0): LiturgicalDayDef[] => {
  const definitions: LiturgicalDayDef[] = [];

  // Thursday of the Lord's Supper (at the Evening Mass).
  definitions.push({
    liturgicalDayId: `thursday_of_the_lords_supper`,
    precedence: Precedence.Triduum_1,
    dateDef: { holyThursday: {}, yearOffset },
    seasons: [{ key: Season.PaschalTriduum, weekOfSeason: 1, dayOfSeason: 0 }],
    periods: [{ key: Period.HolyWeek }],
    dayOfWeek: 4,
    colors: [Color.White],
    i18nDef: [`names:thursday_of_the_lords_supper`],
    fromCalendarId: PROPER_OF_TIME_ID,
  });

  // Friday of the Passion of the Lord.
  definitions.push({
    liturgicalDayId: `friday_of_the_passion_of_the_lord`,
    precedence: Precedence.Triduum_1,
    dateDef: { goodFriday: {}, yearOffset },
    seasons: [{ key: Season.PaschalTriduum, weekOfSeason: 1, dayOfSeason: 1 }],
    periods: [{ key: Period.HolyWeek }],
    dayOfWeek: 5,
    colors: [Color.Red],
    i18nDef: [`names:friday_of_the_passion_of_the_lord`],
    fromCalendarId: PROPER_OF_TIME_ID,
  });

  // Holy Saturday
  definitions.push({
    liturgicalDayId: `holy_saturday`,
    precedence: Precedence.Triduum_1,
    dateDef: { holySaturday: {}, yearOffset },
    seasons: [{ key: Season.PaschalTriduum, weekOfSeason: 1, dayOfSeason: 2 }],
    periods: [{ key: Period.HolyWeek }],
    dayOfWeek: 6,
    colors: [],
    i18nDef: [`names:holy_saturday`],
    fromCalendarId: PROPER_OF_TIME_ID,
  });

  // Easter Sunday of the Resurrection of the Lord.
  definitions.push({
    liturgicalDayId: `easter_sunday`,
    precedence: Precedence.Triduum_1,
    dateDef: { easterSunday: {}, yearOffset },
    isHolyDayOfObligation: true,
    seasons: [
      { key: Season.PaschalTriduum, weekOfSeason: 1, dayOfSeason: 3 },
      { key: Season.EasterTime, weekOfSeason: 1, dayOfSeason: 1 },
    ],
    periods: [{ key: Period.EasterOctave }],
    dayOfWeek: 0,
    colors: [Color.White],
    i18nDef: [`names:easter_sunday`],
    fromCalendarId: PROPER_OF_TIME_ID,
  });

  return definitions;
};
