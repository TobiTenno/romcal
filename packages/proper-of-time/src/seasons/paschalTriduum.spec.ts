import { paschalTriduum } from './paschalTriduum';

describe('paschalTriduum', () => {
  it('should match snapshot', () => {
    expect(paschalTriduum()).toMatchSnapshot();
  });

  it('should match snapshot with year offset to +1', () => {
    expect(paschalTriduum(+1)).toMatchSnapshot();
  });
});
