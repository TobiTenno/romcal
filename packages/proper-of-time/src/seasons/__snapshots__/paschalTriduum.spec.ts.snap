// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`paschalTriduum should match snapshot 1`] = `
[
  {
    "colors": [
      "WHITE",
    ],
    "dateDef": {
      "holyThursday": {},
      "yearOffset": 0,
    },
    "dayOfWeek": 4,
    "fromCalendarId": "proper-of-time",
    "i18nDef": [
      "names:thursday_of_the_lords_supper",
    ],
    "liturgicalDayId": "thursday_of_the_lords_supper",
    "periods": [
      {
        "key": "HOLY_WEEK",
      },
    ],
    "precedence": "01_TRIDUUM",
    "seasons": [
      {
        "dayOfSeason": 0,
        "key": "PASCHAL_TRIDUUM",
        "weekOfSeason": 1,
      },
    ],
  },
  {
    "colors": [
      "RED",
    ],
    "dateDef": {
      "goodFriday": {},
      "yearOffset": 0,
    },
    "dayOfWeek": 5,
    "fromCalendarId": "proper-of-time",
    "i18nDef": [
      "names:friday_of_the_passion_of_the_lord",
    ],
    "liturgicalDayId": "friday_of_the_passion_of_the_lord",
    "periods": [
      {
        "key": "HOLY_WEEK",
      },
    ],
    "precedence": "01_TRIDUUM",
    "seasons": [
      {
        "dayOfSeason": 1,
        "key": "PASCHAL_TRIDUUM",
        "weekOfSeason": 1,
      },
    ],
  },
  {
    "colors": [],
    "dateDef": {
      "holySaturday": {},
      "yearOffset": 0,
    },
    "dayOfWeek": 6,
    "fromCalendarId": "proper-of-time",
    "i18nDef": [
      "names:holy_saturday",
    ],
    "liturgicalDayId": "holy_saturday",
    "periods": [
      {
        "key": "HOLY_WEEK",
      },
    ],
    "precedence": "01_TRIDUUM",
    "seasons": [
      {
        "dayOfSeason": 2,
        "key": "PASCHAL_TRIDUUM",
        "weekOfSeason": 1,
      },
    ],
  },
  {
    "colors": [
      "WHITE",
    ],
    "dateDef": {
      "easterSunday": {},
      "yearOffset": 0,
    },
    "dayOfWeek": 0,
    "fromCalendarId": "proper-of-time",
    "i18nDef": [
      "names:easter_sunday",
    ],
    "isHolyDayOfObligation": true,
    "liturgicalDayId": "easter_sunday",
    "periods": [
      {
        "key": "EASTER_OCTAVE",
      },
    ],
    "precedence": "01_TRIDUUM",
    "seasons": [
      {
        "dayOfSeason": 3,
        "key": "PASCHAL_TRIDUUM",
        "weekOfSeason": 1,
      },
      {
        "dayOfSeason": 1,
        "key": "EASTER_TIME",
        "weekOfSeason": 1,
      },
    ],
  },
]
`;

exports[`paschalTriduum should match snapshot with year offset to +1 1`] = `
[
  {
    "colors": [
      "WHITE",
    ],
    "dateDef": {
      "holyThursday": {},
      "yearOffset": 1,
    },
    "dayOfWeek": 4,
    "fromCalendarId": "proper-of-time",
    "i18nDef": [
      "names:thursday_of_the_lords_supper",
    ],
    "liturgicalDayId": "thursday_of_the_lords_supper",
    "periods": [
      {
        "key": "HOLY_WEEK",
      },
    ],
    "precedence": "01_TRIDUUM",
    "seasons": [
      {
        "dayOfSeason": 0,
        "key": "PASCHAL_TRIDUUM",
        "weekOfSeason": 1,
      },
    ],
  },
  {
    "colors": [
      "RED",
    ],
    "dateDef": {
      "goodFriday": {},
      "yearOffset": 1,
    },
    "dayOfWeek": 5,
    "fromCalendarId": "proper-of-time",
    "i18nDef": [
      "names:friday_of_the_passion_of_the_lord",
    ],
    "liturgicalDayId": "friday_of_the_passion_of_the_lord",
    "periods": [
      {
        "key": "HOLY_WEEK",
      },
    ],
    "precedence": "01_TRIDUUM",
    "seasons": [
      {
        "dayOfSeason": 1,
        "key": "PASCHAL_TRIDUUM",
        "weekOfSeason": 1,
      },
    ],
  },
  {
    "colors": [],
    "dateDef": {
      "holySaturday": {},
      "yearOffset": 1,
    },
    "dayOfWeek": 6,
    "fromCalendarId": "proper-of-time",
    "i18nDef": [
      "names:holy_saturday",
    ],
    "liturgicalDayId": "holy_saturday",
    "periods": [
      {
        "key": "HOLY_WEEK",
      },
    ],
    "precedence": "01_TRIDUUM",
    "seasons": [
      {
        "dayOfSeason": 2,
        "key": "PASCHAL_TRIDUUM",
        "weekOfSeason": 1,
      },
    ],
  },
  {
    "colors": [
      "WHITE",
    ],
    "dateDef": {
      "easterSunday": {},
      "yearOffset": 1,
    },
    "dayOfWeek": 0,
    "fromCalendarId": "proper-of-time",
    "i18nDef": [
      "names:easter_sunday",
    ],
    "isHolyDayOfObligation": true,
    "liturgicalDayId": "easter_sunday",
    "periods": [
      {
        "key": "EASTER_OCTAVE",
      },
    ],
    "precedence": "01_TRIDUUM",
    "seasons": [
      {
        "dayOfSeason": 3,
        "key": "PASCHAL_TRIDUUM",
        "weekOfSeason": 1,
      },
      {
        "dayOfSeason": 1,
        "key": "EASTER_TIME",
        "weekOfSeason": 1,
      },
    ],
  },
]
`;
